drift-idl-generator/src/main/java/com/facebook/drift/idl/generator/ThriftIdlGenerator.java

```diff
@@ -129,6 +131,46 @@ public ThriftIdlGenerator(ThriftIdlGeneratorConfig config, ClassLoader classLoad
 
         this.namespaces = config.getNamespaces();
         this.recursive = config.isRecursive();
+
+        for (String customCodecClassName : config.getCustomCodecs()) {
+            try {
+                Class<?> codecClass = load(customCodecClassName);
+                if (codecClass == null) {
+                    continue;
+                }
+                Object codec;
+
+                try {
+                    codec = codecClass.getDeclaredConstructor(ThriftCatalog.class).newInstance(catalog);
+                }
+                catch (NoSuchMethodException e) {
+                    try {
+                        codec = codecClass.getDeclaredConstructor().newInstance();
+                    }
+                    catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException nestedException) {
+                        throw new ThriftIdlGeneratorException("Can not find constructor for the custom codec.");
+                    }
+                }
+
+                if (codec instanceof ThriftCodec) {
```

Comment:
We should check this first. And if codec is not an instance of `ThriftCodec` -  throw an exception